### PR TITLE
Al fire perk

### PR DESCRIPTION
**Pull request by anjx123**
Implemented basic structure for Buffs, and a foundation perk for future perks

### CODE REVIEW

File: Assets/Scripts/BuffScripts/Buff.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        this.duration = duration;
+    }
+
+    public void decrementTime()
+    {
+        duration = Mathf.Clamp(duration - Time.deltaTime, 0, duration);
+    }
+
+    public float GetDuration()
```
Comment (leo-cf-tian) [inline Assets/Scripts/BuffScripts/Buff.cs:18]: could we consider a property here?

File: Assets/Scripts/BuffScripts/Resistance.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+public class Resistance : Buff
+{
+    private DamageType damageType;
+    private float resistance;
+
+
```
Comment (leo-cf-tian) [inline Assets/Scripts/BuffScripts/Resistance.cs:6]: extraneous newline, might as well get rid of it :P

File: Assets/Scripts/PerkScripts/FireResistance/FireResistance.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
+public class FireResistance : Perk
+{
+    public float resistance = 0.2f;
+
+    public override void OnStart(Player player)
```
Comment (leo-cf-tian) [inline Assets/Scripts/PerkScripts/FireResistance/FireResistance.cs:5]: if it's nothing, you can omit it.

File: Assets/Scripts/PerkScripts/FireResistance/FireResistance.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    }
+
+    public override void OnHit(ref int damage, DamageType damageType, Player player)
+    {
+        if (damageType == DamageType.FIRE)
+            damage = (int)(damage * (1 - resistance));
+    }
+
+    // Start is called before the first frame update
+    void Start()
```
Comment (leo-cf-tian) [inline Assets/Scripts/PerkScripts/FireResistance/FireResistance.cs:17]: you might want to define names etc. at the top for clarity.

File: Assets/Scripts/PerkScripts/Perk.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+public abstract class Perk : MonoBehaviour
+{
+    protected string perkName;
+    protected string description;
+
```
Comment (leo-cf-tian) [inline Assets/Scripts/PerkScripts/Perk.cs:8]: extraneous newlines

File: Assets/Scripts/PerkScripts/Perk.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    {
+
+    }
+
+    public abstract void OnStart(Player player);
+
+    public abstract void OnHit(ref int damage, DamageType damageType, Player player);
+
+
+    public void SetName(string name)
```
Comment (leo-cf-tian) [inline Assets/Scripts/PerkScripts/Perk.cs:28]: same thing about setting names

**Review by leo-cf-tian — CHANGES_REQUESTED** [review]

**Comment by Yikachu** [comment]
wait we didnt merge this??? berm...

File: Assets/Scripts/PlayerScripts/Player.cs [hunk 7]
```diff
@@ -1,21 +1,78 @@
...
+    [Header("Player Stats")]
+    public PlayerHealth playerHealth;
+    public PlayerMovement playerMovement;
+
+    [Header("Buffs and Perks")]
+    public List<Perk> perks = new List<Perk>();
+
+    public List<Buff> buffs = new List<Buff>();
+
+
```
Comment (leo-cf-tian) [inline Assets/Scripts/PlayerScripts/Player.cs:20]: extraneous newline

**Review by leo-cf-tian — APPROVED** [review]
1 style nit, feel free to merge after changing it

**Pull request merged**